### ISSUE TITLE
[Feature] Update CI/GitHub Action in GenericBucket

### ISSUE THREAD

**Issue by LazyGeniusMan** [ScoopInstaller/Scoop#4599]
## Feature Request
Sorry if this is not the right place to ask, I can't open issue in GenericBucket.

#### Is your feature request related to a problem? Please describe.

GenericBucket doesn't have the same feature as known bucket because it's using old yml file and old repo for GitHub Action, is this intentional or you guys forgot to update it?

#### Describe the solution you'd like

Use .github folder from known bucket for GenericBucket and also add `appveyor.yml`(?)

#### I also have some question:
- Does the autoupdate thingy comitted by github-action bot is what you guys called Excavator?
- Does it run purely on github action (dont need to set up anything outside github)?

**Comment by rashil2000** [comment]
The generic bucket needs some rework. Thanks for the reminder. 

I don't currently seem to have access to the repo but we'll probably scrap the current code and start afresh, based on Extras. 

/cc @rasa @issaclin32

**Comment by rashil2000** [comment]
> Does the autoupdate thingy comitted by github-action bot is what you guys called Excavator?

Yes, that automated process is called Excavator in Scoop parlance.

> Does it run purely on github action (dont need to set up anything outside github)?

It used to be run personally by @r15ch13, until when I migrated Main. Extras and Versions to GitHub Actions last month.

**Comment by issaclin32** [comment]
> I don't currently seem to have access to the repo but we'll probably scrap the current code and start afresh, based on Extras.

I have updated permissions for `Scoop/GithubActions`.

**Comment by rashil2000** [comment]
I was saying that we delete the repository, and start from scratch, since it'll no longer be a fork. Also should we use just `Bucket` instead of `GenericBucket` (since it'll be a template, we don't need to say `Bucket-Template`).

On a similar note, might I suggest renaming the `Scoop-GitHubActions` repo to `GitHubActions`, or just `Actions` (it's already under the org so we don't need the `Scoop-` prefix).

**Comment by issaclin32** [comment]
>  I suggest renaming the `Scoop-GitHubActions` repo to `GitHubActions`, or just `Actions` (it's already under the org so we don't need the `Scoop-` prefix).

I just renamed `Scoop-GitHubActions` to `GitHubActions`, and modified the code in `Main`, `Extras`, `Versions` and `nerd-fonts` accordingly.

**Comment by issaclin32** [comment]
> I was saying that we delete the repository, and start from scratch, since it'll no longer be a fork. 

You mean a clean-room implementation? (to avoid licensing problems) , or delete the repo and upload all the content again. (just to get rid of fork status)

**Comment by rashil2000** [comment]
> You mean a clean-room implementation? (to avoid licensing problems)

This.

**Comment by issaclin32** [comment]
I've created the `Bucket` repo. However I still think it would be better to call it `BucketTemplate` or something else, because `Bucket` is a vague name.

**Comment by rashil2000** [comment]
Okay, `BucketTemplate` is also fine. 

(I think you missed this option - https://docs.github.com/en/repositories/creating-and-managing-repositories/creating-a-template-repository)

**Comment by issaclin32** [comment]
> Okay, `BucketTemplate` is also fine.
> 
> (I think you missed this option - https://docs.github.com/en/repositories/creating-and-managing-repositories/creating-a-template-repository)

OK. It is now a template.

**Comment by rashil2000** [comment]
Boilerplate has been added https://github.com/ScoopInstaller/BucketTemplate/commit/7e2900aa66cc24a4c575d0f48847d797520e0f7a

You can delete the ScoopInstaller/GenericBucket repo now.

Also, it'd be good to add BucketTemplate repo to the maintainers team.
